test/distributed/elastic/utils/distributed_test.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import multiprocessing as mp
+import os
+import socket
+import unittest
+from contextlib import closing
+
+from torch.distributed.elastic.utils.distributed import (
+    create_c10d_store,
+    get_free_port,
+    get_socket_with_port,
+)
+
+
+def _create_c10d_store_mp(is_server, server_addr, port, world_size):
+    store = create_c10d_store(is_server, server_addr, port, world_size, timeout=2)
+    if store is None:
+        raise AssertionError()
+
+    store.set(f"test_key/{os.getpid()}", "test_value".encode("UTF-8"))
+
+
+class DistributedUtilTest(unittest.TestCase):
+    def test_create_store_single_server(self):
+        store = create_c10d_store(is_server=True, server_addr=socket.gethostname())
+        self.assertIsNotNone(store)
+
+    def test_create_store_no_port_multi(self):
+        with self.assertRaises(ValueError):
+            create_c10d_store(
+                is_server=True, server_addr=socket.gethostname(), world_size=2
+            )
+
+    def test_create_store_multi(self):
```

Comment:
nit and can be done in followup PR: The rest of distributed tests uses a `MultiProcessTestCase` as base class. Does it make sense if we use that base test class here?

https://github.com/pytorch/pytorch/blob/ac668c55e58e093270be2d72e17d801b83b97e45/torch/testing/_internal/common_distributed.py#L266

Reply:
thanks for pointing this out @mrshenli ! With the awesome changes that @H-Huang  and @cbalioglu  have added to `c10d::Store`, the `utils/distributed.py` file can go away. I'm planning to do this when I upstream the `torchelastic.agent` module since that's the one that is using these util methods.